metals/src/main/scala/scala/meta/internal/metals/codeactions/CreateNewFile.scala

```diff
@@ -0,0 +1,46 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.meta.internal.metals._
+import scala.meta.internal.metals.MetalsEnrichments._
+import org.eclipse.{lsp4j => l}
+import scala.meta.pc.CancelToken
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import java.net.URI
+import java.nio.file.Paths
+
+class CreateNewFile() extends CodeAction {
+  override def kind: String = l.CodeActionKind.QuickFix
+
+  override def contribute(
+      params: l.CodeActionParams,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
+
+    def createNewFile(diagnostic: l.Diagnostic, name: String): l.CodeAction = {
+      val codeAction = new l.CodeAction()
+      codeAction.setTitle(CreateNewFile.title)
+      codeAction.setKind(l.CodeActionKind.QuickFix)
+      codeAction.setDiagnostics(List(diagnostic).asJava)
+      val directory = Paths
+        .get(URI.create(params.getTextDocument().getUri()))
+        .getParent()
+        .toString()
+      codeAction.setCommand(ServerCommands.NewScalaFile.toLSP(List(null, name)))
+      codeAction
+    }
+
+    val codeActions = params.getContext().getDiagnostics().asScala.collect {
+      case d @ ScalacDiagnostic.SymbolNotFound(name)
+          if d.getRange().encloses(params.getRange()) =>
+        createNewFile(d, name)
+    }
+
+    Future.successful(codeActions)
+
+  }
+}
+
+object CreateNewFile {
+  val title = "Create new..."
```

Comment:
`create new symbol 'NAME'`?

Reply:
I thought about adding something like that, but I feel like "symbol" is not very user-friendly (it's very much compiler lingo).

Anything else would require listing out the options ("Create new trait 'NAME'", "Create new class 'NAME'"), but think it would become too heavy.

I feel like "Create new..." is clear when presented in context and it avoids the problem of listing out all the options.

Reply:
I agree "symbol" is not idea but it's used in "rename symbol" and "go to symbol in file" in VS Code so I think it's OK to use it here.

Reply:
You're right, it's used around, at least in VSCode. I'll make the change

Reply:
Done!

![image](https://user-images.githubusercontent.com/691940/77314864-c7f25400-6d06-11ea-84a3-c2e24afa23b2.png)

Reply:
Nice! I think it's a bit more self documenting what this option does with the name. Should we drop the `...` suffix?

Reply:
The `...` is used to convey the fact that the action starts a flow and it doesn't take effect immediately. 

I've seen this pattern used quite often, including in VSCode. See for instance

![image](https://user-images.githubusercontent.com/691940/77316651-17864f00-6d0a-11ea-99d6-2e0aa499ee4b.png)

The commands ending with `...` start a flow that requires further interaction to complete.